sys/include/fido2/ctap/ctap.h

```diff
@@ -0,0 +1,644 @@
+ *
+ * @{
+ *
+ * @file
+ * @brief       Internal FIDO2 CTAP defines, structures and function declarations
+ *
+ * @author      Nils Ollrogge <nils.ollrogge@fu-berlin.de>
+ */
+
+#ifndef FIDO2_CTAP_CTAP_H
+#define FIDO2_CTAP_CTAP_H
+
+#include <stdint.h>
+
+#include "mutex.h"
+#include "cbor.h"
+#include "assert.h"
+#include "crypto/modes/ccm.h"
+
+#include "fido2/ctap.h"
+#include "fido2/ctap/ctap_crypto.h"
+#include "fido2/ctap.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Size of pin auth
+ */
+#define CTAP_PIN_AUTH_SZ 16
+
+/**
+ * @name CTAP methods
+ *
+ * @{
+ */
+#define CTAP_MAKE_CREDENTIAL                0x01
+#define CTAP_GET_ASSERTION                  0x02
+#define CTAP_GET_INFO                       0x04
+#define CTAP_CLIENT_PIN                     0x06
+#define CTAP_RESET                          0x07
+#define CTAP_GET_NEXT_ASSERTION             0x08
+#define CTAP_VENDOR_FIRST                   0x40
+#define CTAP_VENDOR_LAST                    0xBF
+/** @} */
+
+/**
+ * @name CTAP authenticator data option flags
+ *
+ * @{
+ */
+#define CTAP_AUTH_DATA_FLAG_UP     (1 << 0)     /**< user present */
+#define CTAP_AUTH_DATA_FLAG_UV     (1 << 2)     /**< user verified */
+#define CTAP_AUTH_DATA_FLAG_AT     (1 << 6)     /**< attested credential data included */
+#define CTAP_AUTH_DATA_FLAG_ED     (1 << 7)     /**< extension data included */
+/** @} */
+
+/**
+ * @name CTAP version flags
+ *
+ * @{
+ */
+#define CTAP_VERSION_FLAG_FIDO_PRE  0x01
+#define CTAP_VERSION_FLAG_FIDO      0x02
+#define CTAP_VERSION_FLAG_U2F_V2    0x04
+/** @} */
+
+/**
+ * @name CTAP get info response options map CBOR key values
+ *
+ * All options are in the form key-value pairs with string IDs and
+ * boolean values
+ * @{
+ */
+#define CTAP_GET_INFO_RESP_OPTIONS_ID_PLAT       "plat"
+#define CTAP_GET_INFO_RESP_OPTIONS_ID_RK         "rk"
+#define CTAP_GET_INFO_RESP_OPTIONS_ID_CLIENT_PIN "clientPin"
+#define CTAP_GET_INFO_RESP_OPTIONS_ID_UP         "up"
+#define CTAP_GET_INFO_RESP_OPTIONS_ID_UV         "uv"
+/** @} */
+
+/**
+ * @name CTAP get info options flags
+ *
+ * @{
+ */
+#define CTAP_INFO_OPTIONS_FLAG_PLAT         (1 << 0)
+#define CTAP_INFO_OPTIONS_FLAG_RK           (1 << 1)
+#define CTAP_INFO_OPTIONS_FLAG_CLIENT_PIN   (1 << 2)
+#define CTAP_INFO_OPTIONS_FLAG_UP           (1 << 3)
+#define CTAP_INFO_OPTIONS_FLAG_UV           (1 << 4)
+/** @} */
+
+/**
+ * @name CTAP Client PIN request subcommand CBOR key values
+ *
+ * @{
+ */
+#define CTAP_CP_REQ_SUB_COMMAND_GET_RETRIES         0x01
+#define CTAP_CP_REQ_SUB_COMMAND_GET_KEY_AGREEMENT   0x02
+#define CTAP_CP_REQ_SUB_COMMAND_SET_PIN             0x03
+#define CTAP_CP_REQ_SUB_COMMAND_CHANGE_PIN          0x04
+#define CTAP_CP_REQ_SUB_COMMAND_GET_PIN_TOKEN       0x05
+/** @} */
+
+/**
+ * @brief CTAP thread stack size
+ */
+#ifdef CONFIG_FIDO2_CTAP_STACK_SIZE
+#define CTAP_STACKSIZE CONFIG_FIDO2_CTAP_STACK_SIZE
+#else
+#define CTAP_STACKSIZE 15000
```

Comment:
Minor thing, but `CONFIG_FIDO2_CTAP_STACK_SIZE` has a default size of 16k (*sys/fido2/ctap/Kconfig*)